pkg/source/source.go

```diff
@@ -55,6 +55,22 @@ type Source interface {
 	Start(handler.EventHandler, workqueue.RateLimitingInterface, ...predicate.Predicate) error
 }
 
+// NewKindWithCache creates a Source without InjectCache, so that it is assured that the given cache is used
```

Comment:
Nit: the first sentence of this godoc is basically for ppl who know how its implemented and not explaining the purpose which is far more important. It would be nice to just say sth like `NewKindWithCache is used to create a Wach for a different cluster`, then go on with the implementation details like "this is achieved by intentionally not implementing InjectCache" etc. Would be great if someone could create a followup for that.

/lgtm